ui/page/main_page.go

```diff
@@ -666,3 +672,158 @@ func (mp *MainPage) LayoutTopBar(gtx layout.Context) layout.Dimensions {
+		case t.ProposalStatus == wallet.VoteStarted:
+			notification = fmt.Sprintf("Voting has started for proposal with Token: %s", t.Proposal.Token)
+		case t.ProposalStatus == wallet.VoteFinished:
+			notification = fmt.Sprintf("Voting has ended for proposal with Token: %s", t.Proposal.Token)
+		default:
+			notification = fmt.Sprintf("New update for proposal with Token: %s", t.Proposal.Token)
+		}
+		initializeBeepNotification(notification)
+	}
+}
+
+func initializeBeepNotification(n string) {
+	absoluteWdPath, err := GetAbsolutePath()
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	err = beeep.Notify("Decred Godcr Wallet", n, filepath.Join(absoluteWdPath, "ui/assets/decredicons/qrcodeSymbol.png"))
+	if err != nil {
+		log.Info("could not initiate desktop notification, reason:", err.Error())
+	}
+}
+
+// listenForNotifications starts a goroutine to watch for sync updates
+// and update the UI accordingly.
+func (mp *MainPage) listenForNotifications() {
+	// Return if any of the listener is not nill.
+	switch {
+	case mp.SyncProgressListener != nil:
+		return
+	case mp.TxAndBlockNotificationListener != nil:
+		return
+	case mp.ProposalNotificationListener != nil:
+		return
+	}
+
+	mp.SyncProgressListener = listeners.NewSyncProgress()
+	err := mp.WL.MultiWallet.AddSyncProgressListener(mp.SyncProgressListener, MainPageID)
+	if err != nil {
+		log.Errorf("Error adding sync progress listener: %v", err)
+		return
+	}
+
+	mp.TxAndBlockNotificationListener = listeners.NewTxAndBlockNotificationListener()
+	err = mp.WL.MultiWallet.AddTxAndBlockNotificationListener(mp.TxAndBlockNotificationListener, true, MainPageID)
+	if err != nil {
+		log.Errorf("Error adding tx and block notification listener: %v", err)
+		return
+	}
+
+	mp.ProposalNotificationListener = listeners.NewProposalNotificationListener()
+	err = mp.WL.MultiWallet.Politeia.AddNotificationListener(mp.ProposalNotificationListener, MainPageID)
+	if err != nil {
+		log.Errorf("Error adding politeia notification listener: %v", err)
+		return
+	}
+
+	go func() {
+		for {
+			select {
+			case n := <-mp.TxAndBlockNotifChan:
+				switch n.Type {
+				case listeners.NewTransaction:
+					mp.UpdateBalance(false)
+					transactionNotification := mp.WL.Wallet.ReadBoolConfigValueForKey(load.TransactionNotificationConfigKey)
+					if transactionNotification {
+						update := wallet.NewTransaction{
+							Transaction: n.Transaction,
+						}
+						mp.postDesktopNotification(update)
+					}
+					mp.RefreshWindow()
+				case listeners.BlockAttached:
+					beep := mp.WL.Wallet.ReadBoolConfigValueForKey(dcrlibwallet.BeepNewBlocksConfigKey)
+					if beep {
+						_ = beeep.Beep(5, 1)
+					}
+
+					mp.UpdateBalance(false)
+					mp.RefreshWindow()
+				case listeners.TxConfirmed:
+					mp.UpdateBalance(false)
+					mp.RefreshWindow()
+
+				}
+			case notification := <-mp.ProposalNotifChan:
+				// Don't notify on wallet synced event.
```

Comment:
Might read better as
`Post desktop notification for all events except the Synced event.`